pkg/logging/init.go

```diff
@@ -25,42 +26,80 @@ const (
 	initKeyVersion = "version"
 )
 
-type Option func(*loggingConfig)
+type Option func(*loggingConfig) error
 
 // WithOutput sets the log output
 func WithOutput(w io.Writer) Option {
-	return func(lc *loggingConfig) {
+	return func(lc *loggingConfig) error {
 		lc.stdOutput = w
+		return nil
 	}
 }
 
 // WithErrorOutput sets the log output for level Error, Fatal and Panic. For the rest,
 // the default output os.Stdout or the output set by `WithOutput` is chosen
 func WithErrorOutput(w io.Writer) Option {
-	return func(lc *loggingConfig) {
+	return func(lc *loggingConfig) error {
 		lc.errsOutput = w
+		return nil
+	}
+}
+
+var (
+	emptyFilePathError = errors.New("empty filepath provided")
+)
+
+// WithFileOutput sets the log output to a file. The filepath can be one of the following:
+//
+// - stdout: logs will be written to os.Stdout
+// - stderr: logs will be written to os.Stderr
+// - devnull: logs will be discarded
+// - any other filepath: logs will be written to the file
+func WithFileOutput(filepath string) Option {
+	return func(lc *loggingConfig) error {
+		var output io.Writer
+		switch filepath {
+		case "stdout":
```

Comment:
Maybe these could be constants (e.g. `OutputSTDERR = "stderr"` etc.) to make it cleaner. Plus, maybe convert the input to lowercase before comparison, might reduce issues when users interact with it and use `STDERR` (which is quite common)...